lib/buffer.js

```diff
@@ -539,11 +539,29 @@ Buffer.prototype.inspect = function inspect() {
   return '<' + this.constructor.name + ' ' + str + '>';
 };
 
+function assertInteger(val, name) {
+  if (typeof val !== 'undefined' &&
+      (typeof val !== 'number' ||
+       +val !== (val >> 0))) {
```

Comment:
`node::ParseArrayIndex()` uses `Uint32Value()`. So we should use uint32 checks in JS to prevent UB. As this stands, if I pass `-1` it'll be coerced to `4294967295` when it hits native. Also true because you're assigning the value to a `size_t`.